### ISSUE TITLE
feat: Global icon override configurations

### ISSUE THREAD

**Issue by sean-perkins** [ionic-team/ionic-framework#24768]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

Previous of 6.0.5, developers we able to take advantage of a public API from `ionicons` to override usages of lazy loaded icons, with their own custom implementation. 

In 6.0.5, we moved to inlining the SVG assets to reduce the bundle size of the library, effectively breaking developers ability to customize these icon usages at a global level (#24766).

We should introduce new keys to the `IonicConfig` that allows developers to define global overrides for icon usages used in our web components. 

### Describe the Use Case

A developer wants to change the icon used for the search icon for all implementations in their application. They do not want to have to manually specify it for each implementation:

```html
<ion-searchbar search-icon="custom-icon"></ion-searchbar>
```

### Describe Preferred Solution

Developers should have an API to tap into many of the internal icon usages within the web components. The code usage will be slightly different per framework target, but essentially:
```
window.Ionic.config = {
  searchIcon: 'custom-icon'
};
```

The related web component would need to be updated to accept this change:
```
@Prop() searchIcon = config.get('searchIcon') as string;
```

This would apply to most web components. 

### Describe Alternatives

N/A

**Comment by sean-perkins** [comment]
@Spinnenzunge thanks for adding your requirements. 

For input, it looks like we have SASS variables for customizing the SVG that is rendered:
- `$input-ios-input-clear-icon-svg` and `$input-md-input-clear-icon-svg`

This is likely because `ion-icon` wasn't widely implemented into the codebase when those variables were added 4+ years ago. I'll add internal notes for the team to possibly migrate this to align with the other components (and expose a config key to customize that icon).

For `ion-textarea`, I don't seen an implementation for a clear action. There appears to only be support for clearing the textarea on edit.

**Comment by sean-perkins** [comment]
@Spinnenzunge the team spent some time discovering this feature request to better understand the impacts it could have to the eco-system. 

We are currently leaning on alternative implementation approaches, instead of moving forward with this feature. I'd appreciate if you could let us know if any of these patterns cannot work for your use case:

1. Per-implementation: Use the available props for the web component to assign your icon. For example, with `ion-searchbar`, using `searchIcon` to customize the display.
2. Wrapper component: Creating a new component that wraps our web component with the property defaults/icons you would like for your implementation. Use this component in implementations instead of ours directly.

The reason we do not feel comfortable moving forward with this feature right now, is that it creates many new public API configurations for many components, when we have not seen a large demand for this level of customization. We do believe that different icons can introduce a whole new feel to an application and we are open to patterns that solve that. Based on the current ask, we believe component encapsulation (wrapper components) may already address this problem well, without any customization or maintenance from Ionic. 

Thanks!

**Comment by Spinnenzunge** [comment]
Hi @sean-perkins, thanks for your reply.

Component encapsulation is not really a solution in my opinion. Creating a whole new component and propagating every possible input and output property of the wrapped ionic component just to default an icon src attribute is not an elegant option.

I guess we will have to implement the icon individually on every component then. As our need lies mainly with the `ion-searchbar` component, this is manageable to maintain.

Just please, please do NOT remove the config for the `detailIcon`. Having to implement that on every `ion-item` in our application would be devastating.

**Comment by sean-perkins** [comment]
@Spinnenzunge thanks for the follow-up!

I believe you are in an Angular project, correct? This directive should suffice for applying the default icon to all implementations of the search bar. You could alternatively add an input for `searchIcon`, and check if it's manually set and prevent applying your default.

```ts
import { Directive, ElementRef } from '@angular/core';

@Directive({
  // eslint-disable-next-line @angular-eslint/directive-selector
  selector: 'ion-searchbar'
})
export class SearchbarDirective {

  constructor(el: ElementRef) {
    // Your custom icon override 
    el.nativeElement.searchIcon = 'search-circle-outline';
  }

}
```

This will prevent having to implement it in all locations manually (you will just need to declare/export it on a module and import that module). 

We have no intention of removing config values at this time (would be a breaking change to a public API). We would have lots of discovery and input for a v7 release before making a change like that.

**Comment by Spinnenzunge** [comment]
I am in an Angular project yes. I haven't thought about the solution with a directive, I do like that approach.

**Comment by sean-perkins** [comment]
I am going to close out this feature request based on our conversation. I do believe the directive approach is ideal for Angular implementations.

If others come across this issue and are using a framework such as React or Vue, please feel free to open a new issue and we can work through the implementation requirements and use case.

Thanks!

**Comment by Spinnenzunge** [comment]
I just tested creating a directive that sets the detailIcon for every ion-item. Angular does not allow to declare a directive globally. We are using the SCAM (Single Component Angular Module) pattern. This means that we would have to declare the icon styling directive in every module in our app. Which does not really solve my problem to overwrite the detail icon globally. (Also I am not sure what the impact of a directive for every ion-item would be, our app renders a LOT of ion-items...)

So in conclusion I am afraid that this does not solve my problem after all. For the moment I will proceed to work with the icon overwrite that still works for detail icon and just manually set the icon for the search bar.
